### ISSUE TITLE
Sample project crashes

### ISSUE THREAD

**Issue by miwand** [derekblair/ILPDFKit#43]
Using the project on iOS 10.2 makes the project crash -
Getting a lot of these errors:
invalid AES padding.
invalid AES block size.

Finally get this 
2016-12-17 23:17:03.155 ILPDFKitExample[60071:559333] -[__NSDictionaryM addObject:]: unrecognized selector sent to instance 0x608000640720
2016-12-17 23:17:03.164 ILPDFKitExample[60071:559333] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSDictionaryM addObject:]: unrecognized selector sent to instance 0x608000640720'

**Comment by miwand** [comment]
Narrowed down the problem to - Line 171 on ILPDFContainer.m
 [arr addObject:final];

**Comment by StefanoBiasu** [comment]
Seems that radio button form object doesn't have name.

In the Form init

`- (instancetype)initWithFieldDictionary:(ILPDFDictionary *)leaf page:(ILPDFPage *)pg parent:(ILPDFFormContainer *)p`

at line 76

```
for (ILPDFString *obj in [[leaf parentValuesForKey:@"T"] reverseObjectEnumerator]) [nameComponents addObject:[obj textString]];
        _name = [nameComponents componentsJoinedByString:@"."];
```
the method invokation  

`[leaf parentValuesForKey:@"T"] `

returns empty array because of nil parent
and then its name is @""


Also other fields have nil parent and then the ILPDFFormContainer method

`- (void)populateNameTreeNode:(NSMutableDictionary *)node withComponents:(NSArray *)components final:(ILPDFForm *)final`

tries to add different forms in the `_nameTree` variable path @"" mutiple times.

Is it possible that iOS 10.2 made some changes on PDF AcroForm reading?

**Comment by StefanoBiasu** [comment]
It's possible that the issue is related with PDF encryption.
PDF given in project sample has a 128-bit AES encryption. For each form field found in PDF, console prints out at least once the following message:

> invalid AES block size.

Radio buttons are the first occurence of crash, but there could be other cases of crash after radio buttons. 

Tried to feed sample project with:
- PDF not encrypted but with radio buttons
- PDF with non 128-bit AES (but 128-bit RC4) encryption (e.g. http://www.cic.gc.ca/english/passport/forms/pdf/pptc482.pdf) and radio buttons

 in each case no error is given.

**Comment by derekblair** [comment]
Resolved in 6115182fc406dba8e7811413d02c84fcae463850 . The invalid invalid AES block size warnings seem to be new issues introduced in 10.2 relating to Core Graphics PDF calls using in the project.
